interface/Device.tt

```diff
@@ -259,10 +283,29 @@ foreach (var registerMetadata in publicRegisters)
 <#
     }
 
-    if (register.PayloadSpec != null && string.IsNullOrEmpty(register.Converter))
+    if (hasConverter)
     {
 #>
 
+        private static partial <#= register.PayloadInterfaceType #> FormatPayload(<#= interfaceType #> value);
```

Comment:
I might have missed it (or maybe you are validating it in the schema?). But the existence of this method should be conditional on whether the register has write access or not. Unless there is a good reason to leave it in (e.g. generate messages from "fake", or virtual, devices), I would probably simplify pipeline and exclude this method.

Reply:
Remember we always need both anyway to allow simulating "virtual" devices. In this case we do want to generate a `Read` message in software, and for that we need to format the payload.

Reply:
i.e. Parse / Format is different from Read / Write